registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/RouteIntercepter.java

```diff
@@ -0,0 +1,115 @@
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.util.DateUtils;
+import io.mosip.registration.processor.camel.bridge.model.Setting;
+import io.mosip.registration.processor.camel.bridge.processor.TokenGenerationProcessor;
+import io.mosip.registration.processor.core.abstractverticle.MessageBusAddress;
+import io.mosip.registration.processor.core.abstractverticle.WorkflowEventDTO;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.vertx.core.json.JsonObject;
+
+@Component
+public class RouteIntercepter {
+
+	private static final Logger LOGGER = RegProcessorLogger.getLogger(TokenGenerationProcessor.class);
+
+	@Value("${mosip.regproc.camelbridge.pause-settings}")
+	private String settingsString;
+
+	@Value("${mosip.regproc.camelbridge.intercept-hotlisted-key}")
+	private String hotlistedTagKey;
+
+	private String workflowStatusUpdateAddress = MessageBusAddress.WORKFLOW_EVENT_UPDATE_ADDRESS.toString();
+
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	Setting[] settings = null;
+
+	@PostConstruct
+	private void init() throws JsonParseException, JsonMappingException, IOException {
+
+		settings = objectMapper.readValue(settingsString, Setting[].class);
+
+	}
+
+	public List<RouteDefinition> intercept(CamelContext camelContext, RoutesDefinition routes) {
+		List<RouteDefinition> routesDefination = routes.getRoutes();
+		routesDefination.forEach(x -> {
+			try {
+				x.adviceWith(camelContext, new AdviceWithRouteBuilder() {
+
+					@Override
+					public void configure() throws Exception {
+
+						interceptFrom("*").when(new Predicate() {
+
+							@Override
+							public boolean matches(Exchange exchange) {
+
+								String message = (String) exchange.getMessage().getBody();
+								JsonObject json = new JsonObject(message);
+								JsonObject tags = json.getJsonObject("tags");
+								if (tags.containsKey(hotlistedTagKey)) {
+									String fromAddress = exchange.getFromEndpoint().toString();
+									for (Setting setting : settings) {
+										if (Pattern.matches(setting.getFromAddress(), fromAddress) && tags
+												.getString(hotlistedTagKey).equals(setting.getHotlistedReason())) {
+											// workflow dto workfloweventdto
+											WorkflowEventDTO workflowEventDTO = new WorkflowEventDTO();
+											workflowEventDTO.setResumeTimestamp(DateUtils.toISOString(DateUtils
+													.getUTCCurrentDateTime().plusSeconds(setting.getPauseFor())));
+											workflowEventDTO.setRid(json.getString("rid"));
+											workflowEventDTO.setDefaultResumeAction(setting.getDefaultResumeAction());
+											workflowEventDTO.setStatusCode("PAUSED");
```

Comment:
Take this from RegistrationStatusCode.PAUSED.getString()

Reply:
@vishwa-vyom  it is not in core it is in status service adding that as a dependency